### PR TITLE
*: fix data race in ProcessInfo

### PR DESCRIPTION
**Pull request by wshwsh12**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/45126

Problem Summary:

### What is changed and how it works?

Clone some fields from stmtContext to ProcessInfo.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/test/indexmergereadtest/index_merge_reader_test.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ import (
 	"regexp"
```
Comment (hawkingrei) [inline executor/test/indexmergereadtest/index_merge_reader_test.go:20]: Please run ```make bazel_prepare```.

**Review by XuHuaiyu — APPROVED** [review]

**Review by jackysp — COMMENTED** [review]
I just want to know that why https://github.com/pingcap/tidb/pull/39368 could not avoid this data race.

**Comment by wshwsh12** [comment]
> I just want to know that why https://github.com/pingcap/tidb/pull/39368 could not avoid this data race.

https://github.com/pingcap/tidb/pull/39368 avoid data race from the function `(s *SessionVars).InitStatementContext` by calling `s.RefCountOfStmtCtx.TryFreeze()`
If you want to use `RefCountOfStmtCtx` to avoid this data race, we need call `s.RefCountOfStmtCtx.TryFreeze()` in some functions in builder.go file.

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**